services/m3dbnode/main/main.go

```diff
@@ -75,32 +79,29 @@ func main() {
 		}
 	}
 
-	clientOpts, err := server.DefaultClientOptions(id, tchannelNodeAddr, shardSet)
-	if err != nil {
-		log.Fatalf("could not create client options: %v", err)
-	}
-
-	c, err := client.NewClient(clientOpts)
+	cli, err := client.NewClient(server.DefaultClientOptions(topoInit))
```

Comment:
So to avoid having to initialize it up front, I can lazily initialize it and put a lock on the dynamic topology so consequent calls to `.Init()` return the same instance it created the first time. 

This should work well yes?